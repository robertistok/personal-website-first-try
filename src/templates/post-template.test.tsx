import React from "react";
import { useStaticQuery } from "gatsby";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import PostTemplate from "./post-template";
import siteMetadata from "../../jest/__fixtures__/site-metadata";
import markdownRemark from "../../jest/__fixtures__/markdown-remark";

describe("<PostTemplate />", () => {
  const props = {
    data: {
      ...markdownRemark
    }
  };

  beforeEach(() => {
    (useStaticQuery as jest.Mock).mockReturnValue(siteMetadata);
  });

  it("renders", () => {
    shallow(<PostTemplate {...props} />);
  });

  it("renders and matches snapshot", () => {
    const wrapper = shallow(<PostTemplate {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
