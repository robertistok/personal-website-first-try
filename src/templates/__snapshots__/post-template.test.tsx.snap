// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTemplate /> renders and matches snapshot 1`] = `
<Layout
  description="test"
  title="test - Test title"
>
  <Post
    post={
      Object {
        "fields": Object {
          "tagSlugs": Array [
            "/test_0",
            "/test_1",
          ],
        },
        "frontmatter": Object {
          "date": "2016-09-01",
          "description": "test",
          "tags": Array [
            "test_0",
            "test_1",
          ],
          "title": "test",
        },
        "html": "<p>test</p>",
        "id": "test-123",
      }
    }
  />
</Layout>
`;
